src/components/ContactCard/ContactFormField.jsx

```diff
@@ -92,32 +91,57 @@ const ContactFormField = ({
       )}
       {label}
     </label>
-    <ContactFieldInput
-      name={name}
-      type={type}
-      placeholder={placeholder}
-      withLabel={inputWithLabel}
-      required={required}
-      labelPlaceholder={labelPlaceholder}
-    />
+    {!isArray ? (
+      <div className="contact-form__inputs-wrapper">{renderInput(name)}</div>
+    ) : (
+      <FieldArray name={name}>
+        {({ fields }) => (
+          <div className="contact-form__inputs-wrapper">
+            {fields.map((nameWithIndex, index) => {
+              const isLastField = index === fields.length - 1;
+              const canAddField =
+                isLastField && fields.value[index] && fields.value[index][name];
+
+              return (
+                <div className="contact-form__meta-wrapper" key={nameWithIndex}>
+                  {renderInput(`${nameWithIndex}.${name}`)}
+
+                  {!isLastField && (
+                    <button
+                      type="button"
+                      onClick={() => fields.remove(index)}
+                      className="contact-form__meta-button contact-form__meta-button--remove"
+                    >
+                      <Icon icon={IconPlus} />
+                    </button>
+                  )}
+
+                  {canAddField && (
+                    <button
+                      type="button"
+                      onClick={() => fields.push(undefined)}
+                      className="contact-form__meta-button contact-form__meta-button--add"
+                    >
+                      <Icon icon={IconPlus} />
+                    </button>
+                  )}
+                </div>
+              );
```

Comment:
I find the contact of the `fields.map` function a bit hard to read, but I don't think we would gain much from splitting it elsewhere.